### PR TITLE
Add team integration logs API

### PR DESCRIPTION
**Pull request by naormatania**
Add support for https://api.slack.com/methods/team.integrationLogs 
Add unittest to check it works as expected
Also ran `make pr-prep` to check it's ready for review

### CODE REVIEW

File: team.go [hunk 1]
```diff
@@ -43,6 +45,25 @@ type Login struct {
 	Region    string `json:"region"`
 }
 
+type LogResponse struct {
+	Logs   []Log `json:"logs"`
+	Paging `json:"paging"`
+	SlackResponse
+}
+
+type Log struct {
```
Comment (kanata2) [inline team.go:54]: Please add `Channel` field.

**Review by kanata2 — COMMENTED** [review]
Looks good except for the parts I pointed out!

File: team.go (hunk 1 above)
Comment (naormatania) [reply team.go:54]: Thanks for the feedback
done

**Comment by kanata2** [comment]
@naormatania 
When I tried it, I came across strange behavior.
Slack documentation says that the `service_id` field is string, but actual response's one is not string but int... So slack-go/slack could not decode.
I'll confirm Slack support.

**Comment by naorsec** [comment]
@kanata2 has slack support answered about this issue ?

**Comment by kanata2** [comment]
Sorry for delay.
According to support, this is documentation's mistake.

**Comment by naormatania** [comment]
@kanata2 it seems like slack updated their documentation that `service_id` can be both int or string.
I changed PR to handle both cases. let me know what you think. thanks !

**Comment by kanata2** [comment]
@naormatania 
I confirmed support again.
According to support's response, the type of `service_id` is only integer,
so we do not seem to need to consider the string type. :sob:

**Comment by naormatania** [comment]
Thanks for the help @kanata2 ! I changed it to `int` now

**Comment by kanata2** [comment]
@naormatania

thanks!
I found a new undocumented behavior...
It seems that the `user_id` can be 0 (= a number) like following example.

```json
        {
            "user_id": 0,
            "user_name": "slack",
            "date": "1595673313",
            "change_type": "disabled",
            "reason": "channel_archive",
            "service_id": 123456789,
            "service_type": "Incoming WebHooks",
            "scope": "incoming-webhook"
        }
```

File: team.go [hunk 2]
```diff
@@ -43,6 +45,26 @@ type Login struct {
...
+	ServiceID   int    `json:"service_id"`
+	ServiceType string `json:"service_type"`
+	AppID       string `json:"app_id"`
+	AppType     string `json:"app_type"`
+	UserID      string `json:"user_id"`
+	UserName    string `json:"user_name"`
+	Channel     string `json:"channel"`
+	Date        string `json:"date"`
```
Comment (dorsec) [inline team.go:62]:
```go
DateTime   JSONTime `json:"date,string"`
```

File: team.go [hunk 3]
```diff
@@ -43,6 +45,26 @@ type Login struct {
...
+	Paging `json:"paging"`
+	SlackResponse
+}
+
+type Log struct {
+	ServiceID   int    `json:"service_id"`
+	ServiceType string `json:"service_type"`
+	AppID       string `json:"app_id"`
+	AppType     string `json:"app_type"`
+	UserID      string `json:"user_id"`
```
Comment (dorsec) [inline team.go:59]: stringOrInt

**Review by dorsec — CHANGES_REQUESTED** [review]

**Pull request closed without merging**